File: 3859-maximum-product-of-two-digits/maximum-product-of-two-digits.cpp
class Solution {
public:
    int maxProduct(int n) {
        int maxnum = INT_MIN;

        while (n != 0) {
            int digit = n % 10;
            n /= 10;
            int num = n;
            while (num != 0) {
                int innerdigit = num % 10;
                num /= 10;
                maxnum = max(maxnum, digit * innerdigit);
                cout<<maxnum<<endl;
            }
        }

        return maxnum;
    }
};

File: 3859-maximum-product-of-two-digits/maximum-product-of-two-digits.java
class Solution {
    public int maxProduct(int n) {
        int maxnum = Integer.MIN_VALUE;
         int temp = n;

        while (temp != 0) {
            int digit = temp % 10;
            temp /= 10;
            int num = temp;
            while (num != 0) {
                int innerdigit = num % 10;
                num /= 10;
                maxnum = Math.max(maxnum, digit * innerdigit);
            }
        }

        return maxnum;
    }
}